tests/lax_numpy_test.py

```diff
@@ -3166,6 +3166,31 @@ def testReshapeMethod(self, arg_shape, out_shape, dtype):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_inshape={}_outshape={}".format(
+          jtu.format_shape_dtype_string(arg_shape, dtype),
+          jtu.format_shape_dtype_string(out_shape, dtype)),
+       "arg_shape": arg_shape, "out_shape": out_shape, "dtype": dtype}
+      for dtype in default_dtypes
+      for arg_shape, out_shape in [
+          ((2, 2), (2, 4)),
+          ((2, 2), (4, 2)),
+          ((2, 2), (4, 3)),
+          ((3,), (2, 4)),
+          ((3,), (4, 2)),
+          ((3,), (4, 3)),
+          ((2, 2), (0,)),
+          ((2, 2), (0, 2)),
+          ((2, 2), (2, 0)),
+      ]))
```

Comment:
It would be good to also test empty arrays and scalar shapes. How about `for arg_shape, out_shape in zip(all_shapes, all_shapes)`?

Reply:
or if you want to be fancy, `itertools.combinations_with_replaecement(all_shapes, 2)`

Reply:
Or maybe it's `itertools.product(all_shapes, 2)`. Thinking is hard

Reply:
`zip(all_shapes, all_shapes)` would not work here, since we want to test the actual resizing with different shapes. However I can change it to `itertools.product(all_shapes, repeat=2)` if it doesn't matter that it generates quite a lot of test cases. Alternatively I can also remove the parameterization on `dtype` which would reduce the amount of generated tests which is probably fine since there is no `dtype` specific code in `jnp.resize`. Let me know what you prefer.

Reply:
Sorry, I missed your previous comments. I changed it too `itertools.product([0] + array_shapes, repeat=2)`. Unfortunately we can't use  `all_shapes` here, since the numpy reference implementation can't handle `jtu.NUMPY_SCALAR_SHAPE` and  `jtu.PYTHON_SCALAR_SHAPE`.

Reply:
Right, that makes sense.

I think we could probably use `all_shapes` on the left, sinc they are never passed to numpy. so something like: `for arg_shape, out_shape in itertools.product(all_shapes, array_shapes)`. What do you think?

Reply:
Good idea, I changed it.